### PR TITLE
Add `FreeMonitoring` diff tests

### PR DESCRIPTION
**Pull request by noisersup**
Refs FerretDB/FerretDB#1323.

### CODE REVIEW

File: tests/diff/freemonitoring_test.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		slices.Sort(expectedKeys)
+		slices.Sort(actualKeys)
+
+		assert.Equal(t, expectedKeys, actualKeys)
+	})
+
+	t.Run("FerretDB", func(t *testing.T) {
+		t.Parallel()
+
+		expectedRes := bson.D{{"state", "enabled"}, {"message", "monitoring is enabled"}, {"ok", float64(1)}}
```
Comment (rumyantseva) [inline tests/diff/freemonitoring_test.go:57]: So, we don't set `url` on FerretDB?

**Review by rumyantseva — COMMENTED** [review]
The changes look reasonable in general.
A couple of questions:
- We shouldn't update `tests/mongo-go-driver` in such PRs (as we agreed before)
- Do we plan to test `undecided` state? E.g. I had this PR - https://github.com/FerretDB/dance/pull/226/files which is currently assigned on you. What are your plans for it?

**Comment by noisersup** [comment]
> Do we plan to test undecided state? E.g. I had this PR - https://github.com/FerretDB/dance/pull/226/files which is currently assigned on you. What are your plans for it?

@rumyantseva Yeah we can do this on this PR or would you rather want to make it in #226?

**Comment by rumyantseva** [comment]
@noisersup up to you! I'm ok with either option.

File: tests/diff/freemonitoring_test.go (hunk 1 above)
Comment (noisersup) [reply tests/diff/freemonitoring_test.go:57]: No, it's used specifically for mongodb, they use it to generate runtime specific url to access telemetry

**Review by rumyantseva — APPROVED** [review]
LGTM, apart from the fact that we modified mongo-go-driver. 
@noisersup if possible, please revert  mongo-go-driver. changes.

**Review by w84thesun — CHANGES_REQUESTED** [review]
Please fix linters.

**Review by rumyantseva — APPROVED** [review]
LGTM. I guess the changes in ferretdb need to be merged, so this one will pass too.

File: tests/diff/freemonitoring_test.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson"
+	"golang.org/x/exp/slices"
+)
+
+func TestFreeMonitoring(t *testing.T) {
```
Comment (chilagrow) [inline tests/diff/freemonitoring_test.go:26]: Do we want a test for disabling free monitoring?

File: tests/diff/freemonitoring_test.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+			case "state":
+				assert.Equal(t, "enabled", v)
+			case "message":
+				msg := "To see your monitoring data, navigate to the unique URL below. Anyone you share the URL with will also " +
+					"be able to view this page. You can disable monitoring at any time by running db.disableFreeMonitoring()."
+				assert.Equal(t, msg, v)
+			case "ok":
+				assert.Equal(t, float64(1), v)
+			case "url":
+				assert.Regexp(t, "https://cloud.mongodb.com/freemonitoring/cluster/\\w+", v)
```
Comment (AlekSi) [inline tests/diff/freemonitoring_test.go:57]: No need to test that

**Review by AlekSi — CHANGES_REQUESTED** [review]
There is no effective difference between MongoDB and FerretDB. That should not be a diff test.

**Comment by noisersup** [comment]
> There is no effective difference between MongoDB and FerretDB. That should not be a diff test.

@AlekSi: Should we remove this PR then?

**Comment by AlekSi** [comment]
I think so, unless we do have a reason to keep it. But I don’t see it

**Pull request closed without merging**